src/core/tests/type_prop/result.cpp

```diff
@@ -51,3 +55,100 @@ TEST(type_prop, result_layout_invalid) {
+
+    EXPECT_THAT(result->get_output_tensor(0).get_names(), UnorderedElementsAre("input"));
+
+    result->get_output_tensor(0).set_names({"out"});
+    EXPECT_THAT(result->output(0).get_names(), UnorderedElementsAre("out"));
+    EXPECT_THAT(result->get_output_tensor(0).get_names(), UnorderedElementsAre("out"));
+    EXPECT_THAT(a->output(0).get_names(), UnorderedElementsAre("input", "out"));
+    EXPECT_THAT(a->get_output_tensor(0).get_names(), UnorderedElementsAre("input", "out"));
+}
+
+TEST_F(TypePropResultV0Test, change_specific_output_name) {
+    auto a = std::make_shared<Parameter>(element::f32, PartialShape::dynamic());
+    a->get_output_tensor(0).set_names({"input"});
+
+    auto result = make_op(a);
+
+    EXPECT_THAT(result->output(0).get_names(), UnorderedElementsAre("input"));
+
+    result->get_output_tensor(0).set_names({"out"});
+
+    EXPECT_THAT(result->output(0).get_names(), UnorderedElementsAre("out"));
+    EXPECT_THAT(result->get_output_tensor(0).get_names(), UnorderedElementsAre("out"));
+    EXPECT_THAT(a->output(0).get_names(), UnorderedElementsAre("input", "out"));
+    EXPECT_THAT(a->get_output_tensor(0).get_names(), UnorderedElementsAre("input", "out"));
+
+    result->output(0).set_names({"new output"});
+
+    EXPECT_THAT(result->output(0).get_names(), UnorderedElementsAre("new output"));
+    EXPECT_THAT(result->get_output_tensor(0).get_names(), UnorderedElementsAre("new output"));
+    EXPECT_THAT(a->output(0).get_names(), UnorderedElementsAre("input", "new output"));
+    EXPECT_THAT(a->get_output_tensor(0).get_names(), UnorderedElementsAre("input", "new output"));
+}
+
+TEST_F(TypePropResultV0Test, add_specific_output_name) {
+    auto a = std::make_shared<Parameter>(element::f32, PartialShape::dynamic());
+    a->get_output_tensor(0).set_names({"input"});
+
+    auto result = make_op(a);
+
+    EXPECT_THAT(result->output(0).get_names(), UnorderedElementsAre("input"));
+
+    result->output(0).set_names({"out"});
+    result->get_output_tensor(0).add_names({"extra output name", "o1"});
+    result->output(0).add_names({"extra output name", "o2"});
+
+    EXPECT_THAT(result->output(0).get_names(), UnorderedElementsAre("out", "extra output name", "o1", "o2"));
+    EXPECT_THAT(result->get_output_tensor(0).get_names(), UnorderedElementsAre("out", "extra output name", "o1", "o2"));
+    EXPECT_THAT(a->output(0).get_names(), UnorderedElementsAre("input", "out", "extra output name", "o1", "o2"));
+    EXPECT_THAT(a->get_output_tensor(0).get_names(),
+                UnorderedElementsAre("input", "out", "extra output name", "o1", "o2"));
+}
+
+TEST_F(TypePropResultV0Test, preserve_specific_name_on_input_replace) {
+    const auto a = std::make_shared<Parameter>(element::f32, PartialShape::dynamic());
+    a->get_output_tensor(0).set_names({"input a"});
+
+    const auto result = make_op(a);
+    result->output(0).set_names({"out"});
+
+    EXPECT_THAT(result->output(0).get_names(), UnorderedElementsAre("out"));
```

Comment:
What is expected of result->input(0) here? Could you explain / add checks here?

Reply:
```cpp
EXPECT_THAT(result->input(0).get_tensor().get_names(), UnorderedElementsAre("out", "input a"));
```
Check will be added